src/handlers/miner_hbbft_handler.erl

```diff
@@ -297,6 +297,40 @@ handle_message(<<BinMsgIn/binary>>, Index, #state{hbbft = HBBFT, skip_votes = Sk
         {ok, {proposed_skip, ProposedRound, IndexRound}} ->
             Skips1 = Skips#{Index => {ProposedRound, IndexRound}},
             {S0#state{skip_votes = Skips1}, []};
+        {ok, {skip, Round, Sigs}} when Round > CurRound ->
+            %% this is a message containing proof there were enough skip votes for this round
+            %% check we got enough skip messages and they're all valid
+            case length(Sigs) >= (2 * F) + 1 andalso
+                 lists:all(fun({SigIndex, Sig}) ->
+                                   libp2p_crypto:verify(t2b({proposed_skip, Round}), Sig, libp2p_crypto:bin_to_pubkey(lists:nth(SigIndex, S0#state.members)))
+                           end, Sigs) of
+                true ->
+                    lager:info("skipping to ~p", [Round]),
+                    {HBBFT1, ToSend1} =
+                        case hbbft:next_round(HBBFT, Round, []) of
+                            {H1, ok} ->
+                                {H1, []};
+                            {H1, {send, NextMsgs}} ->
+                                {H1, NextMsgs}
+                        end,
+                    {HBBFT2, ToSend2} =
+                        case hbbft:start_on_demand(HBBFT1) of
+                            {H2, already_started} ->
+                                {H2, []};
+                            {H2, {send, Msgs}} ->
+                                {H2, Msgs}
+                        end,
+                    ToSend = fixup_msgs(ToSend1 ++ ToSend2),
+                    %% we retain the skip votes here because we may not succeed and we want the
+                    %% algorithm to converge more quickly in that case
+                    {(state_reset(HBBFT2, S0))#state{skip_votes = S0#state.skip_votes},
+                     [ new_epoch | ToSend]};
```

Comment:
Should we send the skip message to all the nodes who did not sign the skip here?